pkg/controller/core/workload_controller.go

```diff
@@ -313,10 +355,10 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		wlOrig := wl.DeepCopy()
 		if dtCond != nil {
 			apimeta.RemoveStatusCondition(&wl.Status.Conditions, kueue.WorkloadDeactivationTarget)
-		}
-		if wl.Status.RequeueState != nil {
-			wl.Status.RequeueState = nil
-			updated = true
+			// Clear RequeueState when deactivated due to requeuing limit exceeded.
+			if dtCond.Reason == kueue.WorkloadRequeuingLimitExceeded {
+				wl.Status.RequeueState = nil
+			}
```

Comment:
This change looks harmless, but otoh it does not seem needed. I ran all tests and they passed. I suggest to revert unless we have a good case why this is needed